analyzers/maven/maven.go

```diff
@@ -77,24 +80,21 @@ func Discover(dir string, options map[string]interface{}) ([]module.Module, erro
 				return nil
 			}
 
-			submodules, err := maven.Modules(path)
+			submodules, err := maven.Modules(path, dir, checked)
 			if err != nil {
 				log.WithError(err).Debug("could not get modules at path")
 				return err
 			}
 
 			for _, m := range submodules {
 				modules = append(modules, module.Module{
-					Name:        m,
+					Name:        m.Name,
 					Type:        pkg.Maven,
-					BuildTarget: m,
-					Dir:         path,
+					BuildTarget: m.Target,
+					Dir:         dir,
 				})
 			}
-			log.WithField("path", path).Debug("skipping")
-			// Don't continue recursing, because anything else is probably a
-			// subproject.
-			return filepath.SkipDir
```

Comment:
To those who weren't on our call: Removing `SkipDir` doesn't make looking at `<modules>` redundant because of projects (such as [this](https://github.com/google/closure-compiler)) which have a generic `pom.xml` at root but split out their logic into other `.xml` files that are named something other than "pom.xml". The root `pom.xml` file would have, for example: `<module>pom-gwt.xml</module>`. So if we don't look at these `<module>` elements we won't know what the project's depedencies are.